deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -1830,26 +2177,40 @@ publish1(Msg = #basic_message { is_persistent = IsPersistent, id = MsgId },
     UC1 = gb_sets_maybe_insert(NeedsConfirming, MsgId, UC),
     stats({1, 0}, {none, MsgStatus1}, 0,
           State2#vqstate{ next_seq_id = SeqId + 1,
+                          %% @todo Update next_deliver_seq_id if IsDelivered is true
+                          %% @todo Assert when doing the update to be sure there are no "deliver" holes.
```

Comment:
I'm not sure what I meant by this. Spend some time to figure this out, this sounds important.

Reply:
Right this is about not skipping a deliver, every time this should be `next_deliver_seq_id + 1`.

Reply:
Because of how this is incremented there is no need for an assert. However extracting the code that increments into its own function would be good. And then the todo can be dropped.